be/src/exprs/array_functions.cpp

```diff
@@ -1199,4 +1206,213 @@ StatusOr<ColumnPtr> ArrayFunctions::array_slice(FunctionContext* ctx, const Colu
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    auto [array_null, elements, offsets] = unpack_array_column(columns[0]);
+    auto* offsets_ptr = offsets->get_data().data();
+
+    auto result_elements = elements->clone_empty();
+    auto result_offsets = UInt32Column::create();
+    result_offsets->reserve(offsets->size());
+    result_offsets->append(0);
+
+    phmap::flat_hash_set<uint32_t> sets;
+
+    uint32_t hash[elements->size()]{0};
+    elements->fnv_hash(hash, 0, elements->size());
+
+    size_t rows = columns[0]->size();
+    for (auto i = 0; i < rows; i++) {
+        size_t offset = offsets_ptr[i];
+        int64_t array_size = offsets_ptr[i + 1] - offsets_ptr[i];
+
+        if (array_size <= 1) {
+            for (size_t j = 0; j < array_size; j++) {
+                result_elements->append(*elements, offset + j, 1);
+            }
+        } else {
+            // put first
+            size_t start = result_elements->size();
+            result_elements->append(*elements, offset, 1);
+
+            // uint32_t hash[array_size];
+            // elements->fnv_hash(hash, offset, offset + array_size);
+            sets.clear();
+            sets.emplace(hash[offset]);
+
+            for (size_t j = 1; j < array_size; j++) {
+                auto elements_idx = offset + j;
+                // hash check
+                if (!sets.contains(hash[elements_idx])) {
+                    result_elements->append(*elements, elements_idx, 1);
+                    sets.emplace(hash[elements_idx]);
+                    continue;
+                }
+
+                // find same hash
+                bool is_contains = false;
+                for (size_t k = 0; k < j; k++) {
+                    if (hash[k] == hash[elements_idx] && result_elements->equals(start + k, *elements, elements_idx)) {
+                        is_contains = true;
+                        break;
+                    }
+                }
+
+                if (!is_contains) {
+                    result_elements->append(*elements, elements_idx, 1);
+                }
+            }
+        }
+
+        result_offsets->append(result_elements->size());
+    }
+
+    return NullableColumn::create(ArrayColumn::create(std::move(result_elements), result_offsets), array_null);
+}
+
+StatusOr<ColumnPtr> ArrayFunctions::array_reverse_any_types(FunctionContext* ctx, const Columns& columns) {
+    DCHECK_EQ(1, columns.size());
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    auto [array_null, elements, offsets] = unpack_array_column(columns[0]);
+    auto* offsets_ptr = offsets->get_data().data();
+
+    auto result_elements = elements->clone_empty();
+
+    size_t rows = columns[0]->size();
+    for (auto i = 0; i < rows; i++) {
+        size_t offset = offsets_ptr[i];
+        int64_t array_size = offsets_ptr[i + 1] - offsets_ptr[i];
+
+        for (int64_t j = array_size - 1; j >= 0; j--) {
+            result_elements->append(*elements, offset + j, 1);
+        }
```

Comment:
avoid copying for null array with elements?

Reply:
it's will handle by `NullableColumn.append`, don't need special branch